azure-pipelines.yml

```diff
@@ -407,142 +407,131 @@ jobs:
     - bash: ./scripts/ci/test_profile_integration.sh
       displayName: 'Run Integration Test against Profiles'
 
-- job: RunAutomationPython3_6
-  displayName: Run Automation Python 3.6
+- job: AutomationTest
+  displayName: Automation Test (Profile Latest)
   dependsOn: BuildPythonWheel
-  timeoutInMinutes: 90
+  timeoutInMinutes: 120
 
   pool:
     vmImage: 'ubuntu-16.04'
+  strategy:
+    matrix:
+      Python36:
+        python.version: '3.6'
+      Python38:
+        python.version: '3.8'
   steps:
-  - task: UsePythonVersion@0
-    displayName: 'Use Python 3.6'
-    inputs:
-      versionSpec: 3.6
-  - task: DownloadPipelineArtifact@1
-    displayName: 'Download Python Wheels'
-    inputs:
-      TargetPath: '$(Build.ArtifactStagingDirectory)/pypi'
-      artifactName: pypi
-  - bash: |
-      pip install --force-reinstall wheel
-  - task: Bash@3
-    displayName: 'Run Test Automation'
-    inputs:
-      targetType: 'filePath'
-      filePath: ./scripts/ci/test_automation.sh
-    env:
-      ADO_PULL_REQUEST_TARGET_BRANCH: $(System.PullRequest.TargetBranch)
+    - task: UsePythonVersion@0
+      inputs:
+        versionSpec: '$(python.version)'
+      displayName: 'Use Python $(python.version)'
+    - bash: |
+        set -ev
 
-- job: RunAutomationReduced20190301
-  displayName: Run Automation Reduced Python 3.6, Profile 2019-03-01
-  dependsOn: BuildPythonWheel
+        git fetch origin dev
 
-  pool:
-    vmImage: 'ubuntu-16.04'
-  steps:
-  - task: UsePythonVersion@0
-    displayName: 'Use Python 3.6'
-    inputs:
-      versionSpec: 3.6
-  - task: DownloadPipelineArtifact@1
-    displayName: 'Download Python Wheels'
-    inputs:
-      TargetPath: '$(Build.ArtifactStagingDirectory)/pypi'
-      artifactName: pypi
-  - bash: |
-      pip install --force-reinstall wheel
-  - task: Bash@3
-    displayName: 'Run Test Automation'
-    inputs:
-      targetType: 'filePath'
-      filePath: ./scripts/ci/test_automation.sh
-    env:
-      REDUCE_SDK: 'True'
-      AZURE_CLI_TEST_TARGET_PROFILE: '2019-03-01'
-      ADO_PULL_REQUEST_TARGET_BRANCH: $(System.PullRequest.TargetBranch)
+        python -m venv env
+        chmod +x env/bin/activate
+        source env/bin/activate
 
-- job: RunAutomation20180301
-  displayName: Run Automation, Profile 2018-03-01
-  dependsOn: BuildPythonWheel
+        pip install -q azdev
+        azdev --version
```

Comment:
duplicate scripts in different profile, please wrap them in a script file or common variable in yml?

Reply:
I searched and found the ADO template only works for steps not for job. So, I moved those similar steps into a single file `.azure-pipelines/template/automation_test.yml`.